src/routes/dashboard/helpers/contributors.ts

```diff
@@ -0,0 +1,47 @@
+import type { Contributor, GitHubUser } from '../types'
+
+export function getGitHubUsername(gitHubStaff: GitHubUser[], userId: string) {
+  const user =
+    gitHubStaff[gitHubStaff.findIndex((user) => user.id === Number(userId))]
+  if (user?.login) return user.login
+  return ''
+}
+
+export async function getTopContributors(
+  contributors: Contributor[],
+  gitHubStaff: GitHubUser[],
+  n: number
+) {
+  const filtered = contributors
+    .sort((prev, next) => next.answers.length - prev.answers.length)
+    .slice(0, n)
+  const result = []
+  for (const contributor of filtered) {
+    let ghUsername = ''
+    let name = ''
+    if (contributor.githubId) {
+      ghUsername = getGitHubUsername(gitHubStaff, contributor.githubId)
+    }
+    if (ghUsername) {
+      name = await (
+        await fetch(`${import.meta.env.VITE_HOST}/api/github/${ghUsername}`, {
```

Comment:
are we able to call `/api/github/${ghUsername}` directly without the host?

Reply:
Just tried, it threw a type error 'Failed to parse url'

Reply:
Ah this is called server-side, resolving